boltons/dictutils.py

```diff
@@ -693,4 +693,104 @@ def __reversed__(self):
             yield curr[KEY]
             curr = curr[PREV]
 
+
+class SentinelInt(int):
+    pass
+
+
+class FrozenDictBase(tuple):
+
+    def __getitem__(self, key):
+        if isinstance(key, SentinelInt):
+            return tuple.__getitem__(self, key)
+        else:
+            try:
+                return getattr(self, str(key.__hash__()))()
+            except AttributeError:
+                raise KeyError(key)
+            except IndexError:
+                raise KeyError(key)
+
+    def __iter__(self):
+        idx = 0
+        while True:
+            try:
+                yield self[SentinelInt(idx)][0]
+            except IndexError:
+                break
+            idx += 1
+
+    def items(self):
+        """Returns a list of tuples of data"""
+        return [(x, self[x]) for x in self]
+
+    def iteritems(self):
+        """Iterates, returning key, value pairs as a tuple"""
+        idx = 0
+        while True:
+            try:
+                yield self[SentinelInt(idx)][0], self[SentinelInt(idx)][1]
+            except IndexError:
+                break
+            idx += 1
```

Comment:
never tried py2.6. Is still being used at all? I guess if this library is intending to support we do have a problem here. Regarding the subclass - I know. Essentially you are creating a tuple of pairs. Iterating of that tuple will return a tuple, that's a key/value pair. That's exactly what's iterating over a dict return (when using `iteritems` and on Py3 `items`).